### PR TITLE
release-upload.yaml: simplify logic by aggregating steps

### PR DESCRIPTION
**Pull request by smelc**
# Changelog

```yaml
- description: |
    Simplify logic of releasing pipeline
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  - refactoring    # QoL changes
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Follow-up of earlier discussions that were left opened on https://github.com/IntersectMBO/cardano-cli/pull/749

# How to trust this PR

I executed this pipeline manually and it goes through the modified steps nicely: https://github.com/IntersectMBO/cardano-cli/actions/runs/9068498972/job/24915930335

# Checklist

- [X] Commit sequence broadly makes sense and commits have useful messages
- [X] Self-reviewed the diff

### CODE REVIEW

**Comment by smelc** [comment]
Note to reviewers: I will fixup the `Retrigger Hydra ¯ \ _ (ツ) _ / ¯` commit before merging, so that it does not show up in the history :wink:

**Review by carbolymer — APPROVED** [review]

**Comment by smelc** [comment]
I'll merge this one after https://github.com/IntersectMBO/cardano-cli/pull/761

**Pull request merged**
